sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+# Storage Blob Service SDK Migration Guide from 10.x/11.x to 12.x
+
+In this section, we list main changes you need to be aware of when converting your Storage Blob SDK library from Version 10/Version 11 to Version 12.
+
+## Prerequisites
+Java Development Kit (JDK) with version 8 or above
+
+## Reactor and ReactorNetty
+We've updated our asynchronous framework from RxJava to Reactor.
+For more info related to Reactor, please refer to [Project Reactor](https://projectreactor.io/)
+For more info of the motivation behind this major change, please refer to [this guide](https://github.com/Azure/azure-storage-java/blob/master/V12%20Upgrade%20Story.md).
+
+## Converting Core Classes
+Our core asynchronous classes have been replaced, as well as new synchronous counterparts added.
+| Core v10/11 classes  | Equivalent v12 Class | NEW synchronous clients |
+|---|---|---|
+| ServiceURL  |  BlobServiceAsyncClient  | BlobServiceClient  |
+| ContainerURL |  ContainerAsyncClient | ContainerClient |
+| BlobURL    | BlobAsyncClient  |   BlobClient |
+| BlockBlobURL  |  BlockBlobAsyncClient  |  BlockBlobClient  |
+| AppendBlobURL  |  AppendBlobAsyncClient | AppendBlobClient  |
+| PageBlobURL |  PageBlobAsyncClient | PageBlobClient  |
+
+
+## Updated Maven dependency
+Dependency for Blob service:
+```xml
+<dependency>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-storage-blob</artifactId>
+  <version>12.0.0-preview.2</version>
+</dependency>
+```
+
+## Generate SAS token
+SAS token generation has moved from helper classes to the core clients themselves.
+See section [SAS Token](#sastoken) to learn how to use them as a credential.
+Not all fields being set in these samples are necessary.
+At minimum, only an expiry time and permission set are necessary to create a SAS.
+
+In v10/v11, to generate the SAS token, you needed an instance of ServiceSASSignatureValues to fluently build out all it's options. This could them generate SASQueryParameters which contained the actual URL query string to attach to your URL. You also needed to provide container and blob names when necessary, as well as the `SharedKeyCredential` to sign the token with.
```

Comment:
```suggestion
In v10/v11, to generate the SAS token, you needed an instance of ServiceSASSignatureValues to fluently build out all it's options. This could then generate SASQueryParameters which contained the actual URL query string to attach to your URL. You also needed to provide container and blob names when necessary, as well as the `SharedKeyCredential` to sign the token with.
```